src/Microsoft.OData.Core/ODataServiceColectionExtensions.cs

```diff
@@ -0,0 +1,58 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="ODataServiceColectionExtensions.cs" company="Microsoft">
+// Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+namespace Microsoft.Extensions.DependencyInjection
+{
+    using System;
+    using Microsoft.OData;
+    using Microsoft.OData.Edm;
+    using Microsoft.OData.Json;
+    using Microsoft.OData.UriParser;
+
+    /// <summary>
+    /// Extension methods for <see cref="IServiceCollection"/>.
+    /// </summary>
+    public static class ODataServiceColectionExtensions
+    {
+        /// <summary>
+        /// Adds the default OData services to the <see cref="IServiceCollection"/>.
+        /// </summary>
+        /// <param name="services">The <see cref="IServiceCollection"/> to add the services to.</param>
+        /// <returns>The <see cref="IServiceCollection"/> instance itself.</returns>
+        public static IServiceCollection AddDefaultODataServices(this IServiceCollection services)

```

Comment:
Maybe just name it 'AddOData(this IServiceCollection services)'

Reply:
Would this not be a source of conflict or confusion with AspNetCoreOData's `AddOData` extension method?

Reply:
AspNetCoreOData, AddOData() is the extension method on 'IMvcBuilder'

For the IServiceCollection extension method, it's named as "AddODataCore" and its internal. So, I don't think there is a conflict.